frontend/src/components/Dashboard.js

```diff
@@ -1,34 +1,60 @@
-import React from "react";
+import React, {useContext} from "react";
 import styled from "styled-components/macro";
 import Header from "../commons/Header";
 import PrimaryButton from "../commons/PrimaryButton";
 import {useHistory} from "react-router-dom";
+import WeatherDataContext from "../contexts/WeatherDataContext";
+import jwtDecode from "jwt-decode";
 
 export default function Dashboard({latitude, longitude, errorMessage}) {
 
     const history = useHistory();
+    const {weatherData} = useContext(WeatherDataContext);
+    const token = localStorage.getItem('jwtToken');
+    const username = token && jwtDecode(token).sub;
 
     return (
         <PageLayout>
-            <Header headerText={"Ahoi 'test user'"}/>
+            <Header headerText={'Ahoi ' + username}/>
             <DashboardBlock>
-                <section>Current location: {latitude && longitude ? <Bold>Available</Bold> : <Bold>Not Available</Bold>}</section>
+                <section>Location:</section>
                 {latitude && longitude ?
                     <div>
                         <div>Latitude: <Bold>{latitude}</Bold></div>
                         <div>Longitude: <Bold>{longitude}</Bold></div>
                     </div>
                     :
-                    <div className={"error"}>{errorMessage}</div>
+                    errorMessage ?
+                        <div className={"error"}>{errorMessage}</div>
+                        :
+                        <div>Searching position...</div>
```

Comment:
this seems to be a little too encapsulated. Consider using if/else here

Reply:
You're right, it’s not very beautiful. 
But I just tried it with if/else (see code below) and it won’t compile...

```javascript
{
                    if (latitude && longitude) {
                    <div>
                        <div>Latitude: <Bold>{latitude}</Bold></div>
                        <div>Longitude: <Bold>{longitude}</Bold></div>
                    </div>
                }
                    if (errorMessage) {
                    <div className={"error"}>{errorMessage}</div>
                }
                    else {
                    <div>Searching position...</div>
                }
}
```

Reply:
try adding a return 

```
  if (latitude && longitude) {
    return(
    <div>
      <div>Latitude: <Bold>{latitude}</Bold></div>
      <div>Longitude: <Bold>{longitude}</Bold></div>
    </div>)
  }
  if (errorMessage) {
    return <div className={"error"}>{errorMessage}</div>
  }
  else {
    return <div>Searching position...</div>
  }
```

Reply:
I tried till I found out that it is not possible to have if/else within JSX. 
So I guess I'll stick to the ternary expression, since writing the whole statement outside of JSX would be quite complicated.

Reply:
you're absolutely right with if/else within JSX, sorry for that!